src/server/bg_services/replication_server.js

```diff
@@ -19,6 +19,35 @@ async function move_objects_by_type(req) {
     }
 }
 
+/**
+ * delete_objects takes in an array of keys and deletes them from the bucket
+ * @param {*} req request object
+ * @returns 
+ */
+async function delete_objects(req) {
+    const { bucket_name, keys } = req.rpc_params;
+    dbg.log1('replication_server delete_objects: params:', bucket_name, keys);
+    const delete_done_list = [];
+
+    const noobaa_con = cloud_utils.set_noobaa_s3_connection(system_store.data.systems[0]);
+    if (!noobaa_con) throw new Error('noobaa endpoint connection is not started yet...');
+    try {
+        const res = await noobaa_con.deleteObjects({
+            Bucket: bucket_name.unwrap(),
+            Delete: {
+                Objects: keys.map(key => ({ Key: key }))
```

Comment:
here we will have max 1000 keys?

Reply:
It is not being enforced in the code yet. @romayalon should I split the deletes into batches of 1k if there are more than 1k keys?

Reply:
yes, my next suggestion is to keep using the semaphore if we don't have this limit :)

Reply:
Got it. Thank you.

Reply:
@romayalon, done.

Reply:
Marking resolved - addressed.